jujugui/static/gui/src/app/jujulib/index.js

```diff
@@ -41,65 +41,62 @@ var module = module;
      @param redirect {Boolean} Whether or not to redirect on a 401 within
         the bakery.
   */
-  var _makeRequest = function(
-    bakery, path, method, params, callback, parse, redirect) {
-    /**
-       Success callback that attempts to parse any error messages out of the
-       JSON response.
-
-       @param xhr {Object} the XHR response object.
-    */
-    var success = function(xhr) {
-      var data = xhr.target.responseText,
-          error = null;
-
-      if (parse !== false && data) {
-        try {
-          data = JSON.parse(data);
-          if (data.error || data.Error) {
-            error = data.error || data.Error;
-          }
-        } catch(e) {
-          error = e;
-        }
+  const _makeRequest = function(
+    bakery, url, method, params, callback, parse, redirect) {
+    const wrappedCallback = (err, response) => {
+      if (err) {
+        callback(err, null);
+        return;
       }
-      callback(error, data);
-    };
-    /**
-       Failure callback that attempts to parse any error messages out of the
-       JSON response, before invoking the user-specified callback.
-
-       @param xhrOrMessage {Object} the XHR response object; can also be a
-                                    plain error string.
-    */
-    var failure = function(xhrOrMessage) {
-      var data;
+      const text = response.target.responseText;
+      if (!text) {
+        callback(null, null);
+        return;
+      }
+      // TODO frankban: we should really avoid defaulting optional
+      // parameters to true.
```

Comment:
This comment will go away as we get rid of makeResponse. The meaning is that parse is used as an optional parameter (even if in JS all parameters are "optional", in this case we often use makeRequest without passing the parse value), and, if not provided, we assume the response must be parsed. In essence, we assume parse=true. The comment is that we should really try to avoid this behavior, as it is confusing.